int8 type
vector2f depthPosition
vector3f worldPosition